tests/gettext/gettext_potfiles.py

```diff
@@ -24,7 +24,16 @@ def check_potfile(checkfile, potlist):
     global success
 
     potcheckfile = None
+
     if checkfile.endswith(".py"):
+        # Check whether the file should be ignored
+        # - there are some unhandled cases where this check returns false positives,
+        #   such as tests that just use the _() function to translate content of variables
+        #   without marking new strings for translation
+        # - so provide a way to skip the check in such a case
+        if subprocess.call(["grep", "-q", "^# ignore-check: potfiles", checkfile]) == 0:
```

Comment:
I think this condition needs to be moved to the ``if`` above otherwise the ``elif`` branch below will never be entered as it has the same condition.